### PR TITLE
Fix media capture

### PR DESCRIPTION
**Pull request by nehtambe**
**Issue #:**
Media capture is not working in non-prod demo apps (Beta stage) because we are replacing the endpoint value of Chime client with Chime SDK meetings endpoint.

**Description of changes:**
To fix the above issue - 
1. Added a new optional parameter -m to pass Chime SDK meetings endpoint. Whenever an internal team wants to use ChimeSDKMeetings to create regional meetings, they should pass SDK meetings endpoint under this parameter not under -c (chime-endpoint).
2. Update deploy-canary-demo script to pass both Chime and Chime SDK meetings endpoint.

**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
Yes, using the following commands.

npm run deploy -- -r us-east-1 -b chime-sdk-meetings-demo-us-east-1-test -s chime-sdk-meetings-demo-us-east-1-test -o chime-sdk-meetings-iad-demo-test -p chime.aws.internal -c <chime_gamma_endpoint> -m <chime_sdk_meetings_iad_gamma_endpoint> -t -l

npm run deploy -- -r eu-central-1  -b chime-sdk-meetings-demo-eu-central-1 -test -s chime-sdk-meetings-demo-eu-central-1 -test -o chime-sdk-meetings-fra-demo-test -p chime.aws.internal -c <chime_gamma_endpoint> -m <chime_sdk_meetings_fra_gamma_endpoint> -t -l

npm run deploy -- -r ap-southeast-1 -b chime-sdk-meetings-demo-ap-southeast-1-test -s chime-sdk-meetings-demo-ap-southeast-1-test -o chime-sdk-meetings-sin-demo-test -p chime.aws.internal -c <chime_gamma_endpoint> -m <chime_sdk_meetings_sin_gamma_endpoint> -t -l

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/serverless/deploy.js
Comment (devalevenkatesh) [inline demos/serverless/deploy.js:16]: There is a `chimeEndpoint` with same value already, any reasons we need a new one?

File: demos/serverless/deploy.js
Comment (devalevenkatesh) [inline demos/serverless/deploy.js:52]: Suggest changing the name:
AWS SDK Chime Meetings -> AWS Chime SDK Meetings endpoint

**Review by devalevenkatesh — COMMENTED** [review]
Request to check with someone from meetings SDK once.

File: demos/serverless/deploy.js
Comment (nehtambe) [reply demos/serverless/deploy.js:52]: Followed the similar pattern as "--chime-endpoint AWS SDK Chime endpoint"

**Comment by nehtambe** [comment]
> Request to check with someone from meetings SDK once.

Yes, I have asked Harshit to check the PR

**Review by harshitOberoi — CHANGES_REQUESTED** [review]

File: demos/serverless/deploy.js
Comment (nehtambe) [reply demos/serverless/deploy.js:259]: We are using this at other places too, should we replace the conditions using 'ChimeSDKMeetingsEndpoint'?

File: demos/serverless/src/handlers.js
Comment (nehtambe) [reply demos/serverless/src/handlers.js:15]: Didn’t understand the first part. This is a boolean variable and it will always have a default value.

File: demos/serverless/deploy.js
Comment (nehtambe) [reply demos/serverless/deploy.js:259]: I was checking the code, we are using it to determine which client to use. For prod demo app we will not pass ChimeSDKMeetingsEndpoint, so I think we will need "useChimeSDKMeetings" to determine the client.

Comment (harshitOberoi) [reply demos/serverless/deploy.js:259]: so for prod instead of paasing -u  we should pass ChimeSDKMeetingsEndpoint if we want to use meetingsSDK

File: demos/serverless/src/handlers.js
Comment (harshitOberoi) [reply demos/serverless/src/handlers.js:15]: useChimeSDKMeetings  should be true if we pass USE_CHIME_SDK_MEETINGS otherwise it should be false. SO instead of passing -u flag to determine whar chime client to use we should use USE_CHIME_SDK_MEETINGS parameter

Comment (nehtambe) [reply demos/serverless/src/handlers.js:15]: I think you meant set value of constant useChimeSDKMeetings to false, but when CHIME_SDK_MEETINGS_ENDPOINT is passed, set the value to true

File: demos/serverless/deploy.js
Comment (nehtambe) [reply demos/serverless/deploy.js:259]: Okay yes we can do that, in that case it will have same value as CHIME_ENDPOINT

Comment (harshitOberoi) [reply demos/serverless/deploy.js:259]: I think i am fine with that parameter. Just handle a case if -u is true and ChimeSDKMeetingsEndpoint is null use chimeEndpoint.

Comment (nehtambe) [reply demos/serverless/deploy.js:259]: ChimeSDKMeetingsEndpoint will always have default value - 'https://service.chime.aws.amazon.com', we will have to remove the default value in order to check for if the endpoint is passed or not. If we do this, everyone, including external and internal users, will have to pass the ChimeSDKMeetingsEndpoint (-m param) in the deployment command

File: demos/serverless/src/handlers.js
Comment (devalevenkatesh) [inline demos/serverless/src/handlers.js:28]: If this is not satisfied what is the `chimeSDKMeetings` endpoint? Is there a default value of `https://service.chime.aws.amazon.com` to it?

File: script/deploy-canary-demo
Comment (devalevenkatesh) [inline script/deploy-canary-demo:31]: So this is for supporting media capture tests using ChimeSDKMeetings endpoint for creating meeting and Chime's gamma endpoint for calling media capture APIs right?

File: demos/serverless/src/handlers.js
Comment (devalevenkatesh) [reply demos/serverless/src/handlers.js:28]: Also, would it makes sense to just remove the `useChimeSDKMeetings` now that we are adding an explicit `chimeSDKMeetingsEndpoint` URL as new parameter?

Comment (nehtambe) [reply demos/serverless/src/handlers.js:28]: Yes, right, it will be pointing to Prod endpoint

File: script/deploy-canary-demo
Comment (nehtambe) [reply script/deploy-canary-demo:31]: Yes, correct. This is for handling the scenario where we want to create regional meetings using  ChimeSDKMeetings and test a capture using those regional meetings. This is especially for Non-Prod regions, since we have different endpoints.

Comment (nehtambe) [reply script/deploy-canary-demo:31]: So, if someone wants to create regional meetings for non-prod regions, they have to pass the ChimeSDKMeetings endpoint in new parameter ChimeSDKMeetingsEndpoint

**Review by devalevenkatesh — COMMENTED** [review]
LGTM! I will approve once mobile team checks since they use the same demo and hence checking if they need to update something on their side before final approval.

File: demos/serverless/src/handlers.js
Comment (devalevenkatesh) [reply demos/serverless/src/handlers.js:28]: So `useChimeSDKMeetings` is basically true by default and says to use the `ChimeMeetingSDK` for regional endpoints and the `chimeSDKMeetingsEndpoint` actually specifies the regional `endpoint`. Be default, all deployments use the `ChimeMeetingSDK` regional endpoint leaving the canary test demos as we are deploying new demos to test the regional endpoints already.

Discussed offline but the clarification is ^.

File: demos/serverless/src/handlers.js
Comment (ltrung) [inline demos/serverless/src/handlers.js:27]: if `useChimeSDKMeetings` is false but `chimeSDKMeetingsEndpoint` is set then what does that mean? Can we just always respect `chimeSDKMeetingsEndpoint` and if it is set then automatically set `useChimeSDKMeetings` to true?

Comment (nehtambe) [reply demos/serverless/src/handlers.js:27]: In that case, we will be using Global Meetings. Also, we can not do that because we are defaulting value of chimeSDKMeetingsEndpoint to https://service.chime.aws.amazon.com, and we will not have any way to distinguish between when to use Global Meetings Client and Chime SDK meetings client for Prod.

**Review by hokyungh — APPROVED** [review]
Did sanity test with new url. No issue found LGTM

**Review by ltrung — APPROVED** [review]

**Review by harshitOberoi — APPROVED** [review]

**Pull request merged**
